cmd/fyne/internal/commands/mock_runner_test.go

```diff
@@ -0,0 +1,101 @@
+package commands
+
+import (
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockExpectedValue struct {
+	dir   interface{}
+	env   []string
+	osEnv bool
+	args  []string
+}
+
+type mockReturnedValue struct {
+	ret []byte
+	err error
+}
+
+type mockExpectedCall struct {
+	dirSet bool
+	envSet bool
+}
+
+type mockRunOutputValue struct {
+	mockExpectedValue
+	mockReturnedValue
+	mockExpectedCall
+}
+
+type testCommandCall struct {
+	calls []mockRunOutputValue
+	index int
+	t     *testing.T
+}
+
+func (t *testCommandCall) runOutput(args ...string) ([]byte, error) {
+	// Check that we have less than the expected number of call
+	assert.Less(t.t, t.index, len(t.calls))
+	// Check that we have the expected number of parameters for this call
+	assert.Equal(t.t, len(t.calls[t.index].args), len(args))
+	// Check that each argument match our expectation
+	if len(t.calls[t.index].args) == len(args) {
```

Comment:
Isn't this already asserted above?

Reply:
By wrapping this in an if we won't test any arguments and potential failures will be missed

Reply:
> The assert, as far as I could see, don't seem to stop the code and it will lead to a crash if they do not match instead of just reporting an error.

6 lines above there is an assert for the size of `t.calls` against `t.index` which is then used blindly, this would produce the same crash if the assert fails.
If you want the test to stop at a test failure use `require` package instead of `assert` and the safety code can be avoided.

Reply:
Good suggestion. I will update with require.